pkg/controllers/dynakube/activegate.go

```diff
@@ -0,0 +1,59 @@
+package dynakube
+
+import (
+	"context"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta1/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/pkg/clients/dynatrace"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/activegate"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/apimonitoring"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/connectioninfo"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/istio"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/version"
+	"github.com/pkg/errors"
+)
+
+func (controller *Controller) reconcileActiveGate(ctx context.Context, dynakube *dynakube.DynaKube, dtc dynatrace.Client, istioReconciler *istio.Reconciler, connectionReconciler *connectioninfo.Reconciler, versionReconciler *version.Reconciler) error { // nolint: revive
+	if dynakube.NeedsActiveGate() { // TODO: this is not optimal, because this check is in the activegate reconciler as well (to do the cleanup)
+		err := connectionReconciler.ReconcileAG(ctx)
```

Comment:
```suggestion
		err := connectionReconciler.ReconcileActiveGate(ctx)
```

Reply:
Out of consistency because the suggestion was made to rename `ReconcileOA` to `ReconcileOneAgent`

Reply:
done